frontend/public/components/app-contents.tsx

```diff
@@ -87,553 +87,554 @@ const LazyRoute = (props) => (
-                    './cluster-settings/request-header-idp-form' /* webpackChunkName: "request-header-idp-form" */
-                  ).then((m) => m.AddRequestHeaderPage)
-                }
-              />
-              <LazyRoute
-                path="/settings/cluster"
-                loader={() =>
-                  import(
-                    './cluster-settings/cluster-settings' /* webpackChunkName: "cluster-settings" */
-                  ).then((m) => m.ClusterSettingsPage)
-                }
-              />
-
-              <LazyRoute
-                path={'/k8s/cluster/storageclasses/~new/form'}
-                exact
-                loader={() =>
-                  import('./storage-class-form' /* webpackChunkName: "storage-class-form" */).then(
-                    (m) => m.StorageClassForm,
-                  )
-                }
-              />
-
-              <Route path="/k8s/cluster/:plural" exact component={ResourceListPage} />
-              <LazyRoute
-                path="/k8s/cluster/:plural/~new"
-                exact
-                loader={() =>
-                  import('./create-yaml' /* webpackChunkName: "create-yaml" */).then(
-                    (m) => m.CreateYAML,
-                  )
-                }
-              />
-              <Route path="/k8s/cluster/:plural/:name" component={ResourceDetailsPage} />
-              <LazyRoute
-                path="/k8s/ns/:ns/pods/:podName/containers/:name"
-                loader={() => import('./container').then((m) => m.ContainersDetailsPage)}
-              />
-              <LazyRoute
-                path="/k8s/ns/:ns/:plural/~new"
-                exact
-                loader={() =>
-                  import('./create-yaml' /* webpackChunkName: "create-yaml" */).then((m) =>
-                    NamespaceFromURL(m.CreateYAML),
-                  )
-                }
-              />
-              <Route path="/k8s/ns/:ns/:plural/:name" component={ResourceDetailsPage} />
-              <Route path="/k8s/ns/:ns/:plural" exact component={ResourceListPage} />
-
-              <Route path="/k8s/all-namespaces/:plural" exact component={ResourceListPage} />
-              <Route path="/k8s/all-namespaces/:plural/:name" component={ResourceDetailsPage} />
-
-              <LazyRoute
-                path="/error"
-                exact
-                loader={() =>
-                  import('./error' /* webpackChunkName: "error" */).then((m) => m.ErrorPage)
-                }
-              />
-              <Route path="/" exact component={DefaultPage} />
-
-              <LazyRoute
-                loader={() =>
-                  import('./error' /* webpackChunkName: "error" */).then((m) => m.ErrorPage404)
-                }
-              />
-            </Switch>
-          </div>
-        </div>
-      </PageSection>
-    )),
-  ),
-);
+  pageRouteExtensions: getPageRouteExtensions(state),
+}))(({ pageRouteExtensions }) => (
```

Comment:
Won't this cause a re-render here on _every_ state change since `getPageRouteExtensions` returns a new array.

Specifically I forgot the use case for using `memo`. But I believe this was all related to performance to reduce unnecessary re-renders.

@vojtechszocs  Did you do any comparisons?

Reply:
I did some comparisons. It doesn't look like we're worse than before.
However we have regressed (even without this change) from the original performance fix here: https://github.com/openshift/console/pull/1558

Reply:
> Won't this cause a re-render here on _every_ state change since `getPageRouteExtensions` returns a new array.

That's correct :disappointed: 

```ts
plugins.registry
  .getRoutePages()
  .filter((e) => plugins.registry.isExtensionInUse(e, flags))
```

yields a new array, due to `filter` in `getRoutePages()`.  However, the content of such array should stay the same in case of state changes that don't effect `RoutePage` extensions, since we're operating on the same extension objects (`this.extensions` in `ExtensionRegistry`).

I'll update the code to ensure we don't re-render unless `RoutePage` extension required feature flags change.

> @vojtechszocs Did you do any comparisons?

I'll update the code and will do some comparisons.

> I did some comparisons. It doesn't look like we're worse than before.
> However we have regressed (even without this change) from the original performance fix here: #1558

Maybe we can open a bug to track it so we don't forget about it.